ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -486,7 +486,9 @@ KEXT_STATIC int HandleVnodeOperation(
                 KAUTH_VNODE_READ_ATTRIBUTES |
                 KAUTH_VNODE_READ_EXTATTRIBUTES))
         {
-            // Recursively expand directory on rename as user will expect the moved directory to have the same contents as in its original location
+            // Depending on OS version:
```

Comment:
Should this awesome comment go right above the `if (s_osSupportsRenameDetection)` ?

Reply:
I guess I see what it could be here, but I think a comment above the if check would be useful too.  Perhaps we should consider renaming `s_osSupportsRenameDetched` to Mojave or later.

Reply:
`s_osSupportsRenameDetection` was a deliberate choice - the idea being that `s_isMojaveOrNewer` doesn't really answer the question of *why* we need a runtime version check. Whether it's the *right* choice is of course debatable.

Perhaps we could change the comment:

```
   // - (Mojave+, isRename only true for actual renames) Block projected directories from being renamed
   // - (High Sierra, isRename also true for other delete events) Recursively expand directory on rename as user will expect the moved directory to have the same contents as in its original location
```

or something along those lines?